packages/components/src/notice/index.js

```diff
@@ -8,22 +8,70 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { RawHTML } from '@wordpress/element';
+import { RawHTML, useEffect, renderToString } from '@wordpress/element';
+import { speak } from '@wordpress/a11y';
 
 /**
  * Internal dependencies
  */
 import { Button, IconButton } from '../';
 
+/**
+ * Custom hook which announces the message with the given politeness, if a
+ * valid message is provided.
+ *
+ * @param {?string} message    Message to announce.
+ * @param {string}  politeness Politeness in which to announce, one of 'polite'
+ *                             or 'assertive'.
+ */
+function useSpokenMessage( message, politeness ) {
```

Comment:
This hooks re-triggers every time the message and politeness change. Is this what we want? Do you think this hook makes sense in the `a11y` package?

Reply:
> This hooks re-triggers every time the message and politeness change. Is this what we want?

I'm not sure. If the message changes, shouldn't the change be announced? Under what conditions should it, or shouldn't it? It's something I could use some accessibility feedback about, but to avoid risk of information not being announced, I think the current implementation is conservative in that regard. I don't expect the message should be changing very frequently either.

Though, not that I think of it, there may be an issue that `children` "changes" more frequently than expected.

>Do you think this hook makes sense in the a11y package?

Possibly. I intentionally wrote it so that it could be extracted for general usage, though didn't yet take the step of doing so.

Reply:
> Though, now that I think of it, there may be an issue that `children` "changes" more frequently than expected.

In 8ca6db1301387bde58623a6e4fcf830009b5243e, I included a test case to cover this. I was surprised to find that it was not re-speaking the message when providing the "same" message as an element. I would have expected that, given that, generally speaking, `children !== children`. To be more certain about this, I changed the implementation of `useSpokenMessage` to convert the message to a string _before_ the `useEffect` uses it as a dependency, so that it's comparing strings and not elements as having changed (follow-up corrective commit at ddbe697).

>>Do you think this hook makes sense in the a11y package?
>
>Possibly. I intentionally wrote it so that it could be extracted for general usage, though didn't yet take the step of doing so.

Porting this to use in `Snackbar` as well could even further justify this, but I chose not to make the change for now. I'm a bit reluctant to commit to incorporating `@wordpress/element` functionality into `@wordpress/a11y`. For now, the implementation of the hook is pretty trivial, so it's not too concerning to me to have this duplication between `Notice` and `Snackbar`.